rir/src/utils/EnumSet.h

```diff
@@ -92,6 +104,8 @@ class EnumSet {
 
     RIR_INLINE Store to_i() const { return set_; }
 
+    constexpr operator StoreType() const { return set_; }
```

Comment:
what is this?

Reply:
implicit conversion. This function gets called if you pass an EnumSet to a function expecting a StoreType.

I needed this, because I wanted to keep the initial effect part of the template. But a complex type like EnumType cannot be a template parameter. Only a primitive type can.

So when you write:

```Instruction<Effects::Any()>```

Then (at compile time) the `Any` function gets evaluated, this returns an EnumSet, then using the implicit conversion it gets converted to the underlying integer of the bitset. Then finally in the constructor of the instruction, the implicit `Enum(StorageType)` constructor is used to convert it back to an enum set :) yay for zero cost abstractions.